ManalHamdi/assignment1/ManalHamdi_As1_Q1.cc

```diff
@@ -0,0 +1,60 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <ctype.h>
+using namespace std;
+
+bool isAnagramCaseSensitive(string s1, string s2) {
+	if (s1.size() != s2.size())
+		return false;
+
+	sort(s1.begin(), s1.end());
+	sort(s2.begin(), s2.end());
+
+	for (int i = 0; i < s1.size(); i++) {
+		if (s1[i] != s2[i])
+			return false;
+	}
+	return true;
+}
+bool isAnagramCaseInsensitive(string s1, string s2) {
+	if (s1.size() != s2.size())
+		return false;
+	for (int i = 0; i < s1.size(); i++) {
+		s1[i] = tolower(s1[i]);
+		s2[i] = tolower(s2[i]);
+	}
+	return isAnagramCaseSensitive(s1, s2);
```

Comment:
It's good that you're trying to re-use an already implemented function, however this could be potentially inefficient (i.e. if your strings are 1 million characters each as could easily be the case when you work at a company like Google!). Your code would iterate over the strings once to convert them to lower case, then again to sort and again to check for equality.

Consider re-implementing as a single function that would look something like this:

```
bool isAnagram(std::string s1, std::string s2, bool is_case_sensitive) {
    ** if sizes don't match, return **

    std::vector<char> s1_chars;
    std::vector<char> s2_chars;

    for (int i = 0; i < s1.size(); i++) {
        if (is_case_sensitive) { 
            s1_chars.push_back(s1[i]);
            s2_chars.....
        else {
            s1_chars.push_back(tolower(s1[i]));
            s2_chars......
        }
    }

   ** sort both vectors **
   return s1_chars == s2_chars;
}
```

The above solution is not strictly better than yours and has issues, like now instead of just having the two strings you need to allocate double the memory to store the vectors, however I think it's nicer because the code is logically contained within one function. 

Now, if in the future you were asked to also make your anagram functions either ignore or treat digits as letters, all you would have to do is add another parameter to the function signature, e.g. `isAnagram(string s1, string s2, bool is_case_sensitive, bool ignore_digits)` and change the function body accordingly.

With the previous version, you would end up with `isAnagramCaseSensitiveIgnoreDigits`, `isAnagramCaseSensitiveWithDigits`, and so on. This is undesirable spaghetti code that would be a nightmare to maintain :)